packages/hdwallet-xdefi/src/xdefi.ts

```diff
@@ -0,0 +1,340 @@
+    return this.info.ethNextAccountPath(msg);
+  }
+
+  public async ethGetAddress(msg: core.ETHGetAddress): Promise<string | null> {
+    if (this.ethAddress) {
+      return this.ethAddress;
+    }
+    const address = await eth.ethGetAddress(this.provider);
+    if (address) {
+      this.ethAddress = address;
+      return address;
+    } else {
+      this.ethAddress = null;
+      return null;
+    }
+  }
+
+  public async ethSignTx(msg: core.ETHSignTx): Promise<core.ETHSignedTx | null> {
+    const address = await this.ethGetAddress(this.provider);
+    return address ? eth.ethSignTx(msg, this.provider, address) : null;
+  }
+
+  public async ethSendTx(msg: core.ETHSignTx): Promise<core.ETHTxHash | null> {
+    const address = await this.ethGetAddress(this.provider);
+    return address ? eth.ethSendTx(msg, this.provider, address) : null;
+  }
+
+  public async ethSignMessage(msg: core.ETHSignMessage): Promise<core.ETHSignedMessage | null> {
+    const address = await this.ethGetAddress(this.provider);
+    return address ? eth.ethSignMessage(msg, this.provider, address) : null;
+  }
+
+  public async ethVerifyMessage(msg: core.ETHVerifyMessage): Promise<boolean | null> {
+    return eth.ethVerifyMessage(msg, this.provider);
+  }
+
+  public async getDeviceID(): Promise<string> {
+    return "xDeFi:" + (await this.ethGetAddress(this.provider));
+  }
+
+  public async getFirmwareVersion(): Promise<string> {
+    return "xDeFi";
+  }
+}
+
+export class XDeFiHDWalletInfo implements core.HDWalletInfo, core.ETHWalletInfo {
+  readonly _supportsBTCInfo = false;
+  readonly _supportsETHInfo = true;
+  readonly _supportsCosmosInfo = false;
+  readonly _supportsBinanceInfo = false;
+  readonly _supportsRippleInfo = false;
+  readonly _supportsEosInfo = false;
+  readonly _supportsFioInfo = false;
+  readonly _supportsThorchainInfo = false;
+  readonly _supportsSecretInfo = false;
+  readonly _supportsKavaInfo = false;
+  readonly _supportsTerraInfo = false;
+
+  public getVendor(): string {
+    return "XDeFi";
+  }
+
+  public hasOnDevicePinEntry(): boolean {
+    return false;
+  }
+
+  public hasOnDevicePassphrase(): boolean {
+    return true;
+  }
+
+  public hasOnDeviceDisplay(): boolean {
+    return true;
+  }
+
+  public hasOnDeviceRecovery(): boolean {
+    return true;
+  }
+
+  public hasNativeShapeShift(srcCoin: core.Coin, dstCoin: core.Coin): boolean {
+    // It doesn't... yet?
+    return false;
+  }
+
+  public supportsOfflineSigning(): boolean {
+    return false;
+  }
+
+  public supportsBroadcast(): boolean {
+    return true;
+  }
+
+  public describePath(msg: core.DescribePath): core.PathDescription {
+    switch (msg.coin) {
+      case "Ethereum":
+        return eth.describeETHPath(msg.path);
+      default:
+        throw new Error("Unsupported path");
+    }
+  }
+
+  public ethNextAccountPath(msg: core.ETHAccountPath): core.ETHAccountPath | undefined {
+    // TODO: What do we do here?
```

Comment:
This method isn't implemented in anything else, and it's really only in hdwallet for legacy reasons; account paths are (will be?) taken care of by chain-adapters now. (We need to do an API scrub now that we've dropped support for the old beta.shapeshift.com stuff.)